packages/zoltan2/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp

```diff
@@ -2967,6 +2967,11 @@ void AlgMJ<mj_scalar_t, mj_lno_t, mj_gno_t, mj_part_t,
   // max for next concurrentPartCount elements, reduce sum for the last
   // concurrentPartCount elements.
   if(this->comm->getSize()  > 1) {
+    auto host_local_min_max_total =
```

Comment:
Those may be device (CudaUVMSpace) buffers.  Try this:
```c++
auto host_local_min_max_total =
  Kokkos::create_mirror_view(Kokkos::HostSpace(), local_min_max_total);
```

Reply:
With this change, i.e. explicit `HostSpace`, it works. There is a second reduction 
https://github.com/trilinos/Trilinos/blob/8e19d1eccea0b01957d562f82a2b3dfb05d5be5d/packages/zoltan2/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp#L3340
that needs the same change. But then things seems to be running again on lassen :-)

Reply:
Changes implemented.